lib/Controller/ApiTablesController.php

```diff
@@ -71,6 +76,73 @@ public function show(int $id): DataResponse {
 		}
 	}
 
+	/**
+	 * [api v2] Get a table Scheme
+	 *
+	 * @NoAdminRequired
+	 *
+	 * @param int $id Table ID
+	 * @return DataResponse<Http::STATUS_OK, TablesTable, array{}>|DataResponse<Http::STATUS_FORBIDDEN|Http::STATUS_INTERNAL_SERVER_ERROR|Http::STATUS_NOT_FOUND, array{message: string}, array{}>
+	 *
+	 * 200: Scheme returned
+	 * 403: No permissions
+	 * 404: Not found
+	 */
+	public function showScheme(int $id): DataResponse {
+		try {
+			$columns = $this->columnService->findAllByTable($id);
+			$table = $this->service->find($id);
+			$scheme = new TableScheme($table->getTitle(),$table->getEmoji(),$columns, $table->getDescription());
+		 	return new DataResponse($scheme->jsonSerialize());
+		} catch (PermissionError $e) {
+			return $this->handlePermissionError($e);
+		} catch (InternalError $e) {
+			return $this->handleError($e);
+		} catch (NotFoundError $e) {
+			return $this->handleNotFoundError($e);
+		}
+	}
+
+	/**
+	 * @NoAdminRequired
+	 */
+	public function createFromScheme(string $title, string $emoji, string $description, array $columns): DataResponse {
+		try {
+			$table = $this->service->create($title,'custom',$emoji, $description);
+			foreach ($columns as $column) {
+				$this->columnService->create(
+					$this->userId,
+					$table->getId(),
+					null,
+					$column['type'],
+					$column['subtype'],
+					$column['title'],
+					$column['mandatory'],
+					$column['description'],
+
+					$column['textDefault'],
+					$column['textAllowedPattern'],
+					$column['textMaxLength'],
+
+					$column['numberPrefix'],
+					$column['numberSuffix'],
+					$column['numberDefault'],
+					$column['numberMin'],
+					$column['numberMax'],
+					$column['numberDecimals'],
+
+					$column['selectionOptions'] == [] ? '' : $column['selectionOptions'],
+					$column['selectionDefault'],
+
+					$column['datetimeDefault'],
+					[],
+				);};
```

Comment:
Would be good to wrap into a transaction. Also maybe we can move this to the service directly so we could also implement an occ command in the future which can reuse the logic?